src/amo/components/SearchContextCard/index.js

```diff
@@ -117,15 +172,64 @@ export class SearchContextCardBase extends React.Component {
   }
 }
 
-export function mapStateToProps(state) {
+export function mapStateToProps(state: AppState) {
+  const { search } = state;
+  const { filters = {} } = search;
+
+  let currentCategory;
+  let categoryName = null;
+
+  if (
+    search &&
+    filters &&
+    filters.category &&
+    typeof filters.category === 'string'
```

Comment:
I think if there is no category, we wouldn't have to do anything else either which is what this checks. 

I guess I can include it here but then I have to set up a default empty string value to addonType (to make flow happy). This is one of the reasons why I moved it here. 

Just to be clear this is what I have:

```diff
+  let addonType = '';
+
   if (
     search &&
     filters &&
+    filters.addonType &&
+    typeof filters.addonType === 'string' &&
     filters.category &&
     typeof filters.category === 'string'
   ) {
     currentCategory = filters.category;
+
+    addonType = filters.addonType;
+    if (addonType === ADDON_TYPE_THEMES_FILTER) {
+      addonType = ADDON_TYPE_STATIC_THEME;
+    }
   }
```

Reply:
I still think how I have this is okay since the **category** filter is ultimately the determining factor, right? I understand that `addonType` would be present too but you could also have a search with just `addonType`

with that being said, I could move this above as you suggest I just prefer how it was really only defined where it was being used (also I didn't have to assign the default value as a string). Maybe that doesn't matter tho..?

e.g.: https://addons-dev.allizom.org/en-US/firefox/search/?platform=mac&q=test&type=extension